### ISSUE TITLE
Add support for M1 Macs by relying on tensorflow-metal and tensorflow-macos

### ISSUE THREAD

**Issue by gsheni** [alteryx/nlp_primitives#116]
- Per the link here, https://developer.apple.com/metal/tensorflow-plugin/, we can rely on these libaries for support tensorflow on m1 Macs
  - https://pypi.org/project/tensorflow-metal/
  - https://pypi.org/project/tensorflow-macos/
- in `setup.cfg` (if we do #114), we could do the following:
```
is_macos = 'darwin' in _plat
is_arm = platform.processor() == 'arm' or platform.machine() == 'arm64'
```

**Comment by gsheni** [comment]
#### setup.cfg
```
install_requires =
    tensorflow-metal>=0.4.0; platform_machine!='arm' or platform_machine=='arm64'
    tensorflow-macos>=2.8.0; platform_machine!='arm' or platform_machine=='arm64'
    tensorflow>=1.14.0; platform_machine!='arm' or platform_machine!='arm64'
    tensorflow_hub>=0.4.0
```

**Comment by thehomebrewnerd** [comment]
Closed by #117
